### PR TITLE
User page updates

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

User Profile Page
- add program membership type to table (from program service not JWT)
- add user membership tag if they are not a `Collaborator`
- updated `ego-token-utils` minor version
- merged two gql queries for profile page

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [x] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Add copyrights to new files
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/user/index.tsx [hunk 1]
```diff
@@ -70,7 +83,11 @@ export function UserPage({ firstName, lastName }: { firstName: string; lastName:
             <ApiTokenBox apiToken={apiToken} loading={loading} />
           </Column>
           <Column sm={12} md={6}>
-            <ProgramAccessBox isDacoApproved={isDacoApproved} loading={loading} />
+            <ProgramAccessBox
+              isDacoApproved={isDacoApproved}
+              readablePrograms={programs}
```
Comment (anncatton) [inline components/pages/user/index.tsx:88]: missing `readablePrograms` from ProgramAccessBox types

File: components/pages/user/index.tsx [hunk 2]
```diff
@@ -31,21 +31,34 @@ import { useQuery } from '@apollo/react-hooks';
...
+  />
+);
 
-  const { data, loading } = useQuery<ProfileQueryData>(PROFILE);
+export function UserPage({ firstName, lastName }: { firstName: string; lastName: string }) {
+  const { permissions } = useAuthContext();
+  const scopes = getAuthorizedProgramScopes(permissions);
+  const readableProgramShortNames = getReadableProgramShortNames(scopes);
+  const readableProgramDataShortNames = getReadableProgramDataNames(permissions);
+  const readablePrograms = uniq([...readableProgramShortNames, ...readableProgramDataShortNames]);
```
Comment (anncatton) [inline components/pages/user/index.tsx:56]: is this being used now that membership type is coming from the gql request?

File: components/pages/user/ProgramAccessBox/index.tsx [hunk 3]
```diff
@@ -84,6 +82,11 @@ const ProgramTable = (props: { programs: Array<T_ProgramTableProgram> }) => {
             maxWidth: 150,
             Cell: ProgramNameCell,
           },
+          {
+            Header: 'Membership Type',
+            accessor: 'membershipType',
+            Cell: (props) => capitalize(props.value),
+          },
```
Comment (hlminh2000) [inline components/pages/user/ProgramAccessBox/index.tsx:89]: should this be hidden for DCC?

File: components/pages/user/index.tsx (hunk 1 above)
Comment (ciaranschutte) [reply components/pages/user/index.tsx:88]: nice catch! typescript check didn't warn me :'(

**Comment by ciaranschutte** [comment]
@hlminh2000 hide table col and user tag for DCC
@anncatton removed duplicate functionality and fixed types

**Review by hlminh2000 — APPROVED** [review]

**Pull request merged**